packages/pty_linux.go

```diff
@@ -114,8 +114,17 @@ func runWithPty(cmd *exec.Cmd) ([]byte, []byte, error) {
 	}()
 
 	err = cmd.Run()
+	if err != nil {
+		// Yum returns non-zero exit values on non-error conditions.
+		// Only fail on errors which are *not* in this category.
+		if _, ok := err.(*exec.ExitError); !ok {
+			return stdout.Bytes(), stderr.Bytes(), err
+		}
+	}
+
 	if err := tty.Close(); err != nil {
```

Comment:
We still need to close the tty if there is an error in Run

Reply:
Done.